### ISSUE TITLE
no-unreachable breaks with babel parser + do { continue; } while (false);

### ISSUE THREAD

**Issue by addaleax** [eslint/eslint#13030]
**Tell us about your environment**

* **ESLint Version:** 6.8.0
* **Node Version:** master/v14.x-pre
* **npm Version:** 6.13.7

**What parser (default, Babel-ESLint, etc.) are you using?** Babel-ESLint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "env": {
        "browser": true,
        "es6": true
    },
    "globals": {
        "console": "readonly"
    },
    "parser": "babel-eslint",
    "parserOptions": {
        "ecmaVersion": 2018
    },
    "rules": {
        "no-constant-condition": 0,
        "no-unreachable": 2
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

test.js:
```js
do { continue; } while (false);
console.log('This code runs');
```

linted with:
```bash
eslint test.js
```

**What did you expect to happen?**

No `no-unreachable` linter error.

**What actually happened? Please include the actual, raw output from ESLint.**

```
/tmp/eslint-test/test.js
  2:1  error  Unreachable code  no-unreachable

✖ 1 problem (1 error, 0 warnings)
```

**Comment by mdjermanovic** [comment]
Hi @addaleax, thanks for the issue! I can reproduce this false positive with `babel-eslint`.

This looks to me like a problem that should be fixed in `babel-eslint`, but would appreciate another opinion.

I believe it's caused by different order:

* eslint-visitor-keys: `DoWhileStatement: ["body", "test"]`
* Babel: `DoWhileStatement: [ "test", "body"]`

which produces a quite different code path analysis for `do { continue; } while (false);`:

![image](https://user-images.githubusercontent.com/44349756/76439642-f85bf900-63bc-11ea-9b63-e3e13aba4cf0.png)

**Comment by kaicataldo** [comment]
Please make an issue in the [Babel repo](https://github.com/babel/babel).
